WordPress/Classes/ViewRelated/Activity/ActivityListViewController.swift

```diff
@@ -1,20 +1,29 @@
 import Foundation
 import CocoaLumberjack
+import SVProgressHUD
 import WordPressShared
 
 class ActivityListViewController: UITableViewController, ImmuTablePresenter {
 
-    @objc let siteID: Int
-    @objc let service: ActivityServiceRemote
+    let siteID: Int
+    let service: ActivityServiceRemote
+
+    enum Configuration {
+        /// Sequence of increasing delays to apply to the fetch restore status mechanism (in seconds)
+        ///
+        static let delaySequence = [1, 5, 10, 15, 20, 25, 30]
```

Comment:
I think this might be too much waiting? Also, I don't think `IncrementalDelay` was my best effort at naming things, but this delay will repeat forever, so `showErrorFetchingRestoreStatus` never happens now.

I think `DispatchDelayedAction` works better than the timer here, but your initial approach of a single 5 second delay up to 12 times was best.

Reply:
I'm going to add a parameter to the `IncrementalDelay` constructor to allows us to make it repeating or not and this should look better.

Reply:
The problem is that `IncrementalDelay` is designed for an infinite sequence (that's why I say I named it wrong), and will assume that there's always a `next()` item.

Reply:
yeah I got that, I'm looking into how hard would it be to modify it to support both infinite and finite sequences.